conda_rpms/install.py

```diff
@@ -516,18 +643,42 @@ def link(pkgs_dir, prefix, dist, linktype=LINK_HARD, index=None, target_prefix=N
         return
 
     source_dir = join(pkgs_dir, dist)
-    if not run_script(dist, 'pre-link', prefix, target_prefix):
+    if not run_script(prefix, dist, 'pre-link', target_prefix):
         sys.exit('Error: pre-link failed: %s' % dist)
 
     info_dir = join(source_dir, 'info')
     files = list(yield_lines(join(info_dir, 'files')))
     has_prefix_files = read_has_prefix(join(info_dir, 'has_prefix'))
     no_link = read_no_link(info_dir)
 
+    noarch = False
+    # If the distribution is noarch, it will contain a `link.json` file in
+    # the info_dir
+    link_json = join(info_dir, 'link.json')
```

Comment:
This seems shaky. Can't we inspect the metadata in info.json (or whatever it is called)? The bit that says ```noarch: python```?

Reply:
We can check that `noarch: python` exists in the `index.json` of the metadata but we do need to get the information about the entrypoints from the `link.json`